### PR TITLE
Create constraint check ULX command

### PR DESCRIPTION
**Pull request by MOEJ-The-Good**

### CODE REVIEW

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+local function getCheckableEnts( plys )
+    local allEnts = ents.GetAll()
+    local perPlyEnts = {}
+
+    for i = 1, #plys do
+        local ply = plys[i]
+        perPlyEnts[ply] = {}
+    end
+
+    for i = 1, #allEnts do
```
Comment (wrefgtzweve) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:27]: Use `ipair` and table.insert instead

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 2]
```diff
@@ -0,0 +1,129 @@
+CFCUlxCommands.constrainChecker = CFCUlxCommands.constrainChecker or {}
+local cmd = CFCUlxCommands.constrainChecker
+local CATEGORY_NAME = "Utility"
+local IsValid = IsValid
+
+-- Returns the owner of the entity if it is checkable, otherwise returns false
+local function getOwnerIfCheckable( ent )
```
Comment (wrefgtzweve) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:7]: I don't see the need for `IfCheckable`, that's not really what the function is doing. All it does is get the owner.

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    end
+
+    return perPlyEnts
+end
+
+local function countConstraintsForPly( ownedEnts )
+    local constraintCounts = {}
+    local totalConstraints = 0
+
+    for i2 = 1, #ownedEnts do
```
Comment (wrefgtzweve) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:47]: use ipairs loop

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    return owner
+end
+
+-- Returns a lookup table from players to the entities they own
+local function getCheckableEnts( plys )
+    local allEnts = ents.GetAll()
+    local perPlyEnts = {}
+
+    for i = 1, #plys do
```
Comment (wrefgtzweve) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:22]: `ipairs`....

File: lua/ulx/modules/sh/cfc_constraint_checker.lua [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                Total = 3,
+                Weld = 3,
+            },
+        }
+--]]
+local function countConstraints( plys )
+    local perPlyEnts = getCheckableEnts( plys )
+    local perPlyConstraints = {}
+
+    for i = 1, #plys do
```
Comment (wrefgtzweve) [inline lua/ulx/modules/sh/cfc_constraint_checker.lua:84]: `ipairs`

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 1 above)
Comment (MOEJ-The-Good) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:27]: Replaced with `ipair`, but `table.insert` only takes a number for the index the value is inputted in. Unless there is a way to make `perPlyEnts[ply] = {}` in the `table.insert` format, I think the way it is, is the only way.

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 3 above)
Comment (MOEJ-The-Good) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:47]: 👍

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 5 above)
Comment (MOEJ-The-Good) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:84]: 👍

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 4 above)
Comment (MOEJ-The-Good) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:22]: 👍

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 1 above)
Comment (wrefgtzweve) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:27]: i mean for `ownedEnts[#ownedEnts + 1] = ent` not for `perPlyEnts`

File: lua/ulx/modules/sh/cfc_constraint_checker.lua (hunk 2 above)
Comment (wrefgtzweve) [reply lua/ulx/modules/sh/cfc_constraint_checker.lua:7]: yeah, even getOwner() would be more logical

**Comment by wrefgtzweve** [comment]
Would it be possible to have colored prints? the console text is quite dry and hard to read right now. Especially when you try and run it on the entire server at once its painful to parse.

image

**Review by wrefgtzweve — APPROVED** [review]

**Pull request merged**